// n8n-nodes-evolution-api/dist/nodes/EvolutionApi/execute/messages/sendReaction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.sendReaction = sendReaction;
const n8n_workflow_1 = require("n8n-workflow");
const evolutionRequest_1 = require("../evolutionRequest");
async function sendReaction(ef) {
    try {
        const instanceName = ef.getNodeParameter('instanceName', 0);
        const remoteJid = ef.getNodeParameter('remoteJid', 0);
        const messageId = ef.getNodeParameter('messageId', 0);
        const fromMe = ef.getNodeParameter('fromMe', 0);
        const reaction = ef.getNodeParameter('reaction', 0);
        if (!reaction) {
            const errorData = {
                success: false,
                error: {
                    message: 'Reação inválida',
                    details: 'É necessário fornecer um emoji para a reação',
                    code: 'INVALID_REACTION',
                    timestamp: new Date().toISOString(),
                },
            };
            return {
                json: errorData,
                error: errorData,
            };
        }
        const body = {
            key: {
                remoteJid,
                fromMe,
                id: messageId,
            },
            reaction,
        };
        const requestOptions = {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
            },
            uri: `/message/sendReaction/${instanceName}`,
            body,
            json: true,
        };
        const response = await (0, evolutionRequest_1.evolutionRequest)(ef, requestOptions);
        return {
            json: {
                success: true,
                data: response,
            },
        };
    }
    catch (error) {
        const errorData = {
            success: false,
            error: {
                message: error.message.includes('Could not get parameter')
                    ? 'Parâmetros inválidos ou ausentes'
                    : 'Erro ao enviar reação',
                details: error.message.includes('Could not get parameter')
                    ? 'Verifique se todos os campos obrigatórios foram preenchidos corretamente'
                    : error.message,
                code: error.code || 'UNKNOWN_ERROR',
                timestamp: new Date().toISOString(),
            },
        };
        if (!ef.continueOnFail()) {
            throw new n8n_workflow_1.NodeOperationError(ef.getNode(), error.message, {
                message: errorData.error.message,
                description: errorData.error.details,
            });
        }
        return {
            json: errorData,
            error: errorData,
        };
    }
}
