// n8n-nodes-evolution-api/dist/nodes/EvolutionApi/execute/instance/deleteInstance.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.deleteInstance = deleteInstance;
const n8n_workflow_1 = require("n8n-workflow");
const evolutionRequest_1 = require("../evolutionRequest");
async function deleteInstance(ef) {
    try {
        const instanceName = ef.getNodeParameter('instanceName', 0);
        const options = {
            method: 'DELETE',
            uri: `/instance/delete/${instanceName}`,
            json: true,
        };
        const response = await (0, evolutionRequest_1.evolutionRequest)(ef, options);
        return {
            json: {
                success: true,
                data: response,
            },
        };
    }
    catch (error) {
        const errorData = {
            success: false,
            error: {
                message: error.message.includes('Could not get parameter')
                    ? 'Par칙metros inv치lidos ou ausentes'
                    : 'Erro ao deletar inst칙ncia',
                details: error.message.includes('Could not get parameter')
                    ? 'Verifique se todos os campos obrigat칩rios foram preenchidos corretamente'
                    : error.message,
                code: error.code || 'UNKNOWN_ERROR',
                timestamp: new Date().toISOString(),
            },
        };
        if (!ef.continueOnFail()) {
            throw new n8n_workflow_1.NodeOperationError(ef.getNode(), error.message, {
                message: errorData.error.message,
                description: errorData.error.details,
            });
        }
        return {
            json: errorData,
            error: errorData,
        };
    }
}
