// n8n-nodes-evolution-api/dist/nodes/EvolutionApi/execute/instance/setPresence.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.setPresence = setPresence;
const n8n_workflow_1 = require("n8n-workflow");
const evolutionRequest_1 = require("../evolutionRequest");
async function setPresence(ef) {
    try {
        const instanceName = ef.getNodeParameter('instanceName', 0);
        const presence = ef.getNodeParameter('presence', 0);
        const options = {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
            },
            uri: `/instance/setPresence/${instanceName}`,
            body: {
                'presence': presence
            },
            json: true,
        };
        const response = await (0, evolutionRequest_1.evolutionRequest)(ef, options);
        return {
            json: {
                success: true,
                data: response,
            },
        };
    }
    catch (error) {
        const errorData = {
            success: false,
            error: {
                message: error.message.includes('Could not get parameter')
                    ? 'Parâmetros inválidos ou ausentes'
                    : 'Erro ao definir presença da instância',
                details: error.message.includes('Could not get parameter')
                    ? 'Verifique se todos os campos obrigatórios foram preenchidos corretamente'
                    : error.message,
                code: error.code || 'UNKNOWN_ERROR',
                timestamp: new Date().toISOString(),
            },
        };
        if (!ef.continueOnFail()) {
            throw new n8n_workflow_1.NodeOperationError(ef.getNode(), error.message, {
                message: errorData.error.message,
                description: errorData.error.details,
            });
        }
        return {
            json: errorData,
            error: errorData,
        };
    }
}
