// n8n-nodes-evolution-api/dist/nodes/EvolutionApi/properties/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.evolutionNodeProperties = void 0;
const events_fields_1 = require("./events.fields");
const instances_fields_1 = require("./instances.fields");
const integrations_fields_1 = require("./integrations.fields");
const messages_fields_1 = require("./messages.fields");
const events_operations_1 = require("./events.operations");
const instances_operations_1 = require("./instances.operations");
const integrations_operations_1 = require("./integrations.operations");
const messages_operations_1 = require("./messages.operations");
const groups_fields_1 = require("./groups.fields");
const groups_operations_1 = require("./groups.operations");
const chat_fields_1 = require("./chat.fields");
const chat_operations_1 = require("./chat.operations");
const profile_fields_1 = require("./profile.fields");
const profile_operations_1 = require("./profile.operations");
const resourcesOptions = {
    displayName: 'Recurso',
    name: 'resource',
    type: 'options',
    noDataExpression: true,
    options: [
        {
            name: 'Instancia',
            value: 'instances-api',
        },
        {
            name: 'Mensagem',
            value: 'messages-api',
        },
        {
            name: 'Grupo',
            value: 'groups-api',
        },
        {
            name: 'Chat',
            value: 'chat-api',
        },
        {
            name: 'Perfil',
            value: 'profile-api',
        },
        {
            name: 'Evento',
            value: 'events-api',
        },
        {
            name: 'Integração',
            value: 'integrations-api',
        },
    ],
    default: 'instances-api',
};
exports.evolutionNodeProperties = [
    resourcesOptions,
    instances_operations_1.instancesOperationsOptions,
    messages_operations_1.messagesOperationsOptions,
    groups_operations_1.groupsOperations,
    events_operations_1.eventsOperationsOptions,
    integrations_operations_1.integrationsOperationsOptions,
    profile_operations_1.profileOperationsOptions,
    chat_operations_1.chatOperations,
    ...instances_fields_1.instancesFields,
    ...messages_fields_1.messagesFields,
    ...groups_fields_1.groupsFields,
    ...events_fields_1.eventsFields,
    ...integrations_fields_1.integrationsFields,
    ...chat_fields_1.chatFields,
    ...profile_fields_1.profileFields,
];
