// n8n-nodes-evolution-api/dist/nodes/EvolutionApi/properties/instances.operations.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.instancesOperationsOptions = void 0;
exports.instancesOperationsOptions = {
    displayName: 'Operação',
    name: 'operation',
    type: 'options',
    noDataExpression: true,
    displayOptions: {
        show: {
            resource: ['instances-api'],
        },
    },
    options: [
        {
            name: 'Criar Instancia',
            action: 'Criar instancia',
            description: 'Cria uma nova Instancia',
            value: 'instance-basic',
        },
        {
            name: 'Conectar Instancia',
            action: 'Conectar instancia',
            description: 'Gera a conexão de uma Instancia (QR ou Base64)',
            value: 'instance-connect',
        },
        {
            name: 'Buscar Instancia',
            action: 'Buscar instancia',
            description: 'Busca e lista as Instancias criadas',
            value: 'fetch-instances',
        },
        {
            name: 'Definir Comportamento',
            action: 'Definir comportamento',
            description: 'Define o comportamento da instancia',
            value: 'instance-settings',
        },
        {
            name: 'Definir Presença',
            action: 'Definir presen a',
            description: 'Define a presença na instancia',
            value: 'set-presence',
        },
        {
            name: 'Definir/Buscar Proxy',
            action: 'Proxy',
            description: 'Define um Proxy na instancia',
            value: 'set-proxy',
        },
        {
            name: 'Reiniciar Instancia',
            action: 'Reiniciar instancia',
            description: 'Reinicia o socket da Instancia',
            value: 'restart-instance',
        },
        {
            name: 'Desconectar Instancia',
            action: 'Desconectar instancia',
            description: 'Desconecta o WhatsApp da Instancia',
            value: 'logout-instance',
        },
        {
            name: 'Deletar Instancia',
            action: 'Deletar instancia',
            description: 'Deleta uma Instancia',
            value: 'delete-instance',
        },
    ],
    default: 'instance-basic',
};
