// n8n-nodes-evolution-api/dist/nodes/EvolutionApi/properties/integrations.operations.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.integrationsOperationsOptions = void 0;
exports.integrationsOperationsOptions = {
    displayName: 'Operação',
    name: 'operation',
    type: 'options',
    noDataExpression: true,
    displayOptions: {
        show: {
            resource: ['integrations-api'],
        },
    },
    options: [
        {
            name: 'Chatwoot',
            action: 'Chatwoot',
            description: 'Define/Busca integração com Chatwoot',
            value: 'chatwoot',
        },
        {
            name: 'Evolution Bot',
            action: 'Evolution bot',
            description: 'Controla a integração com Evolution Bot',
            value: 'evolution-bot',
        },
        {
            name: 'Typebot',
            action: 'Typebot',
            description: 'Controla a integração com Typebot',
            value: 'typebot',
        },
        {
            name: 'Dify',
            action: 'Dify',
            description: 'Controla a integração com Dify',
            value: 'difyBot',
        },
        {
            name: 'Flowise',
            action: 'Flowise',
            description: 'Controla a integração com Flowise',
            value: 'flowiseBot',
        },
    ],
    default: 'chatwoot',
};
