// n8n-nodes-evolution-api/dist/nodes/EvolutionApi/properties/messages.operations.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.messagesOperationsOptions = void 0;
exports.messagesOperationsOptions = {
    displayName: 'Operação',
    name: 'operation',
    type: 'options',
    noDataExpression: true,
    displayOptions: {
        show: {
            resource: ['messages-api'],
        },
    },
    options: [
        {
            name: 'Enviar Texto',
            action: 'Enviar texto',
            description: 'Envia mensagem de Texto',
            value: 'send-text',
        },
        {
            name: 'Enviar Imagem',
            action: 'Enviar imagem',
            description: 'Envia mensagem de Imagem',
            value: 'send-image',
        },
        {
            name: 'Enviar Video',
            action: 'Enviar video',
            description: 'Enviar mensagem de Video',
            value: 'send-video',
        },
        {
            name: 'Enviar Audio',
            action: 'Enviar audio',
            description: 'Enviar mensagem de Audio',
            value: 'send-audio',
        },
        {
            name: 'Enviar Documento',
            action: 'Enviar documento',
            description: 'Enviar mensagem com Documento',
            value: 'send-document',
        },
        {
            name: 'Enviar Enquete',
            action: 'Enviar enquete',
            description: 'Envia uma Enquete de até 12 opções',
            value: 'send-poll',
        },
        {
            name: 'Enviar Contato',
            action: 'Enviar contato',
            description: 'Envia um contato no whatsapp',
            value: 'send-contact',
        },
        {
            name: 'Enviar Lista',
            action: 'Enviar lista',
            description: 'Envia uma lista de opções interativa',
            value: 'send-list',
        },
        {
            name: 'Enviar Botões',
            action: 'Enviar bot es',
            description: 'Envia mensagem com botões interativos',
            value: 'send-buttons',
        },
        {
            name: 'Enviar PIX',
            action: 'Enviar PIX',
            description: 'Envia botão de pagamento PIX',
            value: 'send-pix',
        },
        {
            name: 'Enviar Status',
            action: 'Enviar status',
            description: 'Publicar um Status/Stories',
            value: 'send-stories',
        },
        {
            name: 'Reagir Mensagem',
            action: 'Reagir mensagem',
            description: 'Adiciona uma reação em uma mensagem',
            value: 'send-reaction',
        },
    ],
    default: 'send-text',
};
