// n8n-nodes-evolution-api/dist/nodes/EvolutionApi/execute/groups/createGroup.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.createGroup = createGroup;
const n8n_workflow_1 = require("n8n-workflow");
const evolutionRequest_1 = require("../evolutionRequest");
async function createGroup(ef) {
    try {
        const instanceName = ef.getNodeParameter('instanceName', 0);
        const subject = ef.getNodeParameter('subject', 0);
        const description = ef.getNodeParameter('description', 0);
        const participantsString = ef.getNodeParameter('participants', 0);
        const participants = participantsString.split(',').map(number => number.trim());
        const body = {
            subject,
            description,
            participants,
        };
        const requestOptions = {
            method: 'POST',
            uri: `/group/create/${instanceName}`,
            body,
            json: true,
        };
        const response = await (0, evolutionRequest_1.evolutionRequest)(ef, requestOptions);
        return {
            json: {
                success: true,
                data: response,
            },
        };
    }
    catch (error) {
        const errorData = {
            success: false,
            error: {
                message: error.message,
                details: 'Erro ao criar grupo',
                code: error.code || 'UNKNOWN_ERROR',
                timestamp: new Date().toISOString(),
            },
        };
        if (!ef.continueOnFail()) {
            throw new n8n_workflow_1.NodeOperationError(ef.getNode(), error.message, {
                message: errorData.error.message,
                description: errorData.error.details,
            });
        }
        return {
            json: errorData,
            error: errorData,
        };
    }
}
