// n8n-nodes-evolution-api/dist/nodes/EvolutionApi/execute/groups/toggleEphemeral.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.toggleEphemeral = toggleEphemeral;
const n8n_workflow_1 = require("n8n-workflow");
const evolutionRequest_1 = require("../evolutionRequest");
async function toggleEphemeral(ef) {
    try {
        const instanceName = ef.getNodeParameter('instanceName', 0);
        const groupJid = ef.getNodeParameter('groupJid', 0);
        const expiration = ef.getNodeParameter('expiration', 0);
        const body = {
            expiration,
        };
        const requestOptions = {
            method: 'POST',
            uri: `/group/toggleEphemeral/${instanceName}?groupJid=${groupJid}`,
            body,
            json: true,
        };
        const response = await (0, evolutionRequest_1.evolutionRequest)(ef, requestOptions);
        return {
            json: {
                success: true,
                data: response,
            },
        };
    }
    catch (error) {
        const errorData = {
            success: false,
            error: {
                message: error.message,
                details: 'Erro ao configurar mensagens temporárias',
                code: error.code || 'UNKNOWN_ERROR',
                timestamp: new Date().toISOString(),
            },
        };
        if (!ef.continueOnFail()) {
            throw new n8n_workflow_1.NodeOperationError(ef.getNode(), error.message, {
                message: errorData.error.message,
                description: errorData.error.details,
            });
        }
        return {
            json: errorData,
            error: errorData,
        };
    }
}
