// n8n-nodes-evolution-api/dist/credentials/EvolutionApi.credentials.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EvolutionApi = void 0;
class EvolutionApi {
    constructor() {
        this.name = 'evolutionApi';
        this.displayName = 'Evolution API';
        this.documentationUrl = 'https://doc.evolution-api.com/pt';
        this.properties = [
            {
                displayName: 'Server Url',
                name: 'server-url',
                type: 'string',
                default: '',
                placeholder: 'https://exemplo.com',
                description: 'Digite a URL completa da sua Evolution API (ex: https://api.exemplo.com)',
            },
            {
                displayName: 'ApiKey',
                name: 'apikey',
                type: 'string',
                default: '',
                typeOptions: {
                    password: true,
                },
                description: 'Digite a ApiKey da Instancia ou Global da sua Evolution API',
            },
        ];
        this.authenticate = {
            type: 'generic',
            properties: {
                headers: {
                    apikey: '={{$credentials.apikey}}',
                },
            },
        };
        this.test = {
            request: {
                baseURL: '={{$credentials["server-url"]}}',
                url: '={{$credentials["server-url"].endsWith("/") ? "/erro" : "/instance/fetchInstances"}}',
                method: 'GET',
                headers: {
                    apikey: '={{$credentials.apikey}}',
                },
            },
        };
    }
}
exports.EvolutionApi = EvolutionApi;
