// n8n-nodes-evolution-api/dist/nodes/EvolutionApi/execute/profile/updatePrivacySettings.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.updatePrivacySettings = updatePrivacySettings;
const n8n_workflow_1 = require("n8n-workflow");
const evolutionRequest_1 = require("../evolutionRequest");
async function updatePrivacySettings(ef) {
    try {
        const instanceName = ef.getNodeParameter('instanceName', 0);
        const readreceipts = ef.getNodeParameter('readreceipts', 0);
        const profile = ef.getNodeParameter('profile', 0);
        const status = ef.getNodeParameter('status', 0);
        const online = ef.getNodeParameter('online', 0);
        const last = ef.getNodeParameter('last', 0);
        const groupadd = ef.getNodeParameter('groupadd', 0);
        const body = {
            readreceipts,
            profile,
            status,
            online,
            last,
            groupadd,
        };
        const requestOptions = {
            method: 'POST',
            uri: `/chat/updatePrivacySettings/${instanceName}`,
            body,
            json: true,
        };
        const response = await (0, evolutionRequest_1.evolutionRequest)(ef, requestOptions);
        return {
            json: {
                success: true,
                data: response,
            },
        };
    }
    catch (error) {
        const errorData = {
            success: false,
            error: {
                message: error.message,
                details: 'Erro ao atualizar configurações de privacidade',
                code: error.code || 'UNKNOWN_ERROR',
                timestamp: new Date().toISOString(),
            },
        };
        if (!ef.continueOnFail()) {
            throw new n8n_workflow_1.NodeOperationError(ef.getNode(), error.message, {
                message: errorData.error.message,
                description: errorData.error.details,
            });
        }
        return {
            json: errorData,
            error: errorData,
        };
    }
}
